test/Validate.Layout.cpp

```diff
@@ -46,30 +47,98 @@ using std::vector;
 
 using ::testing::HasSubstr;
 
-using pred_type = function<bool(int)>;
+using pred_type = function<spv_result_t(int)>;
 using ValidateLayout =
     spvtest::ValidateBase<tuple<int, tuple<string, pred_type, pred_type>>,
                           SPV_VALIDATE_LAYOUT_BIT>;
 
+// TODO(umar): probably should move this to a central location
+ostream& operator<<(ostream& os, const spv_result_t& err) {
```

Comment:
I thought we couldn't use `to_string` because the Android compiler does not support it? Are you suggesting I should create a function like `to_string` for `spv_result_t` instead of a stream operator? perhaps a `spvResultToString`.